website/js/index.js

```diff
@@ -588,15 +705,196 @@ var VariantDetail = React.createClass({
+        let groupsEmpty = 0;
+        let totalRowsEmpty = 0;
+
+        const groupTables = _.map(groups, ({groupTitle, innerCols }) => {
+            let rowsEmpty = 0;
+
+            // now map the group's columns to a list of row objects
+            const rows = _.map(innerCols, ({prop, title}) => {
+                let rowItem;
+
+                if (prop === "Protein_Change") {
+                    title = "Abbreviated AA Change";
+                }
+                if (variant[prop] !== null) {
+                    if (prop === "Gene_Symbol") {
+                        rowItem = <i>{variant[prop]}</i>;
+                    }
+                    else if (prop === "URL_ENIGMA") {
+                        if (variant[prop].length) {
+                            rowItem = <a target="_blank" href={variant[prop]}>link to multifactorial analysis</a>;
+                        }
+                    } else if (prop === "Assertion_method_citation_ENIGMA") {
+                        rowItem = <a target="_blank" href="https://enigmaconsortium.org/library/general-documents/">Enigma Rules version Mar 26, 2015</a>;
+
+                        // this will be used in All Data display
+                        /*
+                        } else if (prop == "Source_URL") {
+                         var url_count = 0;
+                         rowItem = _.map(variant[prop].split(','), url => (url.length != 0) && (<span><a key={"Source_URL"+(url_count++)} target="_blank" href={url}>link to multifactorial analysis ({url_count})</a><br /></span>));
+                        */
+                    } else if (prop === "Source_URL") {
+                        if (variant[prop].startsWith("http://hci-exlovd.hci.utah.edu")) {
+                            rowItem = <a target="_blank" href={variant[prop].split(',')[0]}>link to multifactorial analysis</a>;
+                        }
+                    } else if (prop === "Comment_on_clinical_significance_ENIGMA" || prop === "Clinical_significance_citations_ENIGMA") {
+                        const pubmed = "http://ncbi.nlm.nih.gov/pubmed/";
+                        rowItem = _.map(variant[prop].split(/PMID:? ?([0-9]+)/), piece =>
+                            (/^[0-9]+$/.test(piece)) ? <a target="_blank" href={pubmed + piece}>PMID: {piece}</a> : piece );
+                    } else if (prop === "HGVS_cDNA") {
+                        rowItem = variant[prop].split(":")[1];
+                    } else if (prop === "HGVS_Protein") {
+                        rowItem = variant[prop].split(":")[1];
+                    } else if (prop === "Date_last_evaluated_ENIGMA" && !isEmptyField(variant[prop])) {
+                        rowItem = moment(variant[prop], "MM/DD/YYYY").format("DD MMMM YYYY");
+                    } else {
+                        // FIXME: make this less of a hack in the future
+                        // make sure commas wrap
+                        if (variant[prop]) {
```

Comment:
can you wrap all of this normalization logic into a function for easier use/management?